plugins/woocommerce/tests/e2e-pw/tests/merchant/command-palette.spec.js

```diff
@@ -0,0 +1,181 @@
+const { test, expect } = require( '@playwright/test' );
+const wcApi = require( '@woocommerce/woocommerce-rest-api' ).default;
+
+const goToPostEditor = async ( { page } ) => {
+	await page.goto( 'wp-admin/post-new.php' );
+
+	const welcomeModalVisible =
+		await test.step( 'Check if the Welcome modal appeared', async () => {
+			return await page
+				.getByRole( 'heading', {
+					name: 'Welcome to the block editor',
+				} )
+				.isVisible();
+		} );
+
+	if ( welcomeModalVisible ) {
+		await test.step( 'Welcome modal appeared. Close it.', async () => {
+			await page
+				.getByRole( 'document' )
+				.getByRole( 'button', { name: 'Close' } )
+				.click();
+		} );
+	} else {
+		await test.step( 'Welcome modal did not appear.', async () => {
+			// do nothing.
+		} );
+	}
+};
+
+const clickOnCommandPaletteOption = async ( { page, optionName } ) => {
+	// Press `Ctrl` + `K` to open the command palette.
+	await page.keyboard.press( 'Control+K' );
+
+	await page.getByLabel( 'Command palette' ).fill( optionName );
+
+	// Click on the relevant option.
+	const option = page.getByRole( 'option', {
+		name: optionName,
+		exact: true,
+	} );
+	await expect( option ).toBeVisible();
+	option.click();
+};
+
+test.describe( 'Use Command Palette commands', () => {
+	test.use( { storageState: process.env.ADMINSTATE } );
+
+	let productId;
+
+	test.beforeAll( async ( { baseURL } ) => {
+		const api = new wcApi( {
+			url: baseURL,
+			consumerKey: process.env.CONSUMER_KEY,
+			consumerSecret: process.env.CONSUMER_SECRET,
+			version: 'wc/v3',
+		} );
+		await api
+			.post( 'products', {
+				name: 'Product to search',
+				type: 'simple',
+				regular_price: '12.99',
+			} )
+			.then( ( response ) => {
+				productId = response.data.id;
+			} );
+	} );
+
+	test.afterAll( async ( { baseURL } ) => {
+		const api = new wcApi( {
+			url: baseURL,
+			consumerKey: process.env.CONSUMER_KEY,
+			consumerSecret: process.env.CONSUMER_SECRET,
+			version: 'wc/v3',
+		} );
+		await api.delete( `products/${ productId }`, {
+			force: true,
+		} );
+	} );
+
+	test( 'can use the "Add new product" command', async ( { page } ) => {
+		await goToPostEditor( { page } );
+
+		await clickOnCommandPaletteOption( {
+			page,
+			optionName: 'Add new product',
+		} );
+		await page.waitForLoadState( 'networkidle' );
```

Comment:
AFAIK, `networkidle` is discouraged, see https://github.com/microsoft/playwright/issues/2515#issuecomment-724163391 and https://github.com/microsoft/playwright/issues/22897#issuecomment-1540452266. I think we should follow that suggestion as it may be flaky, the event fires after 500ms, and there isn't any guarantee that there isn't any script loading after that.

Reply:
Oooh, you are right. Thanks for the suggestion. I removed it in 398a9e24abcf41452eb75d4567c149c5643a8f30.